src/force-app/main/default/classes/UserSelector.cls

```diff
@@ -0,0 +1,15 @@
+public with sharing class UserSelector {
+    @TestVisible
+    private static UserSelector instance;
+
+    public static UserSelector getInstance() {
+        if (instance == null) {
+            instance = new UserSelector();
+        }
+        return instance;
+    }
+    
+    public Set<Id> getAllInactiveUserIds() {
+        return new Map<Id, User>([SELECT Id FROM User WHERE isActive = false]).keySet();
```

Comment:
This query is too heavy. Instead you should query only records you're interested at during specific transaction.

15 mins to fix.